src/realm-settings/ProfilesTab.tsx

```diff
@@ -15,36 +15,51 @@ import { useTranslation } from "react-i18next";
 import { useAdminClient } from "../context/auth/AdminClient";
 import { useConfirmDialog } from "../components/confirm-dialog/ConfirmDialog";
 import { useRealm } from "../context/realm-context/RealmContext";
-import { cellWidth, IFormatterValueType } from "@patternfly/react-table";
 import { useAlerts } from "../components/alert/Alerts";
 import { Link } from "react-router-dom";
-import type ClientProfilesRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientProfilesRepresentation";
 import "./RealmSettingsSection.css";
+import type ClientPolicyExecutorRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientPolicyExecutorRepresentation";
+
+type ClientProfile = {
+  description?: string;
+  executors?: ClientPolicyExecutorRepresentation[];
+  name?: string;
+  global?: boolean;
+};
 
 export const ProfilesTab = () => {
   const { t } = useTranslation("realm-settings");
   const adminClient = useAdminClient();
   const { realm: realmName } = useRealm();
   const { addAlert, addError } = useAlerts();
-  const [profiles, setProfiles] = useState<ClientProfilesRepresentation>();
+  const [profiles, setProfiles] = useState<ClientProfile[]>();
   const [show, setShow] = useState(false);
 
   const loader = async () => {
-    const clientProfiles = await adminClient.clientPolicies.listProfiles({
+    const allProfiles = await adminClient.clientPolicies.listProfiles({
       realm: realmName,
       includeGlobalProfiles: true,
     });
-    setProfiles(clientProfiles);
-    const allClientProfiles = clientProfiles.globalProfiles?.concat(
-      clientProfiles.profiles!
+
+    const globalProfiles = allProfiles.globalProfiles?.map(
+      (globalProfiles) => ({
+        ...globalProfiles,
+        global: true,
+      })
     );
 
+    const profiles = allProfiles.profiles?.map((profiles) => ({
+      ...profiles,
+      global: false,
+    }));
+
+    const allClientProfiles = globalProfiles?.concat(profiles!);
```

Comment:
Since `profiles` theoretically could be `undefined` it might be better to fall back to default here rather than using the non-null assertion operator.
```suggestion
    const allClientProfiles = globalProfiles?.concat(profiles ?? []);
```